### PR TITLE
chore: Updating Azure NPM to v1.0.30

### PR DESCRIPTION
**Pull request by jaer-tsun**
**Reason for Change**:

Updates to the latest release of Azure NPM:
- removing telemetry sidecar from NPM
- set NPM resource limits

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jaer-tsun** [comment]
splitting the NPM and CNI version updates

File: packer/install-dependencies.sh [hunk 1]
```diff
@@ -62,8 +62,8 @@ installGPUDrivers
 echo "  - nvidia-docker2 nvidia-container-runtime" >> ${VHD_LOGS_FILEPATH}
 
 VNET_CNI_VERSIONS="
+1.0.30
 1.0.29
-1.0.28
```
Comment (jackfrancis) [inline packer/install-dependencies.sh]: Let's keep any changes to the `VNET_CNI_VERSIONS` to the Azure CNI PR

**Comment by jackfrancis** [comment]
@jaer-tsun Thank you for creating two separate PRs for Azure CNI and NPM! :)

File: packer/install-dependencies.sh (hunk 1 above)
Comment (jaer-tsun) [reply packer/install-dependencies.sh]: oops, fixed it

**Comment by jackfrancis** [comment]
@jaer-tsun run `make build` to generate artifacts, and then commit the resultant generated code

**Comment by jackfrancis** [comment]
@jaer-tsun FYI I rebased on top of master to get a recent Jenkinsfile update so that I could run a specific Azure NPM-enabled cluster configuration test against this PR prior to merging

**Comment by jaer-tsun** [comment]
@jackfrancis thanks!

**Comment by jaer-tsun** [comment]
@jackfrancis I merged with master, not sure if I missed anything though

**Comment by jaer-tsun** [comment]
@CecileRobertMichon just changed the versions in this file pkg/api/k8s_versions.go

the previous unit tests failed pretty quickly though, I may be missing something

**Comment by CecileRobertMichon** [comment]
@jaer-tsun I believe there are some references to v1.0.29 in unit tests that you'll need to update to v1.0.30 to fix the failures

**Comment by jaer-tsun** [comment]
@CecileRobertMichon I've just check for v1.0.18, v1.0.27-v1.0.29 and it looks like I've updated everything other than the *.txt files

**Comment by CecileRobertMichon** [comment]
Took a peak at the [logs](https://github.com/Azure/aks-engine/pull/2443/checks?check_run_id=351608018), this is what is failing:

```
--- FAIL: TestSetAddonsConfig (0.01s)
    --- FAIL: TestSetAddonsConfig/Azure_Network_Policy_addon_enabled_-_1.16_upgrade (0.00s)
        addons_test.go:2891: expected addon azure-npm-daemonset to have container MemoryRequests  at at Containers index 0, got 20Mi
    --- FAIL: TestSetAddonsConfig/Azure_Network_Policy_addon_enabled_-_1.16 (0.00s)
        addons_test.go:2891: expected addon azure-npm-daemonset to have container MemoryRequests  at at Containers index 0, got 20Mi
    --- FAIL: TestSetAddonsConfig/Azure_Network_Policy_addon_enabled (0.00s)
        addons_test.go:2891: expected addon azure-npm-daemonset to have container MemoryRequests  at at Containers index 0, got 20Mi
```
You can see pipeline logs by clicking "Details" next to the PR check.

**Comment by jaer-tsun** [comment]
I added the requests and limits for npm. Where to I set the container requests and limits?

**Comment by jackfrancis** [comment]
What is the intention here w/ respect to removing the telemetry sidecar? Is that only for the pre-1.16 npm spec? NPM currently has a different spec for clusters running >= 1.16 and cluster running older versions.

**Comment by jaer-tsun** [comment]
We were sending data through the sidecar from v1.0.27-v1.0.29, but we recently removed it in v1.0.30.

**Comment by CecileRobertMichon** [comment]
@jaer-tsun I believe you need to modify `TestSetAddonsConfig` in addons_test.go

**Comment by jackfrancis** [comment]
Done in #2472

**Comment by jaer-tsun** [comment]
thanks @jackfrancis!

**Pull request closed without merging**